kola/etcdregistry.go

```diff
@@ -12,37 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package etcdtests
+package kola
 
-import "github.com/coreos/mantle/platform"
+import "github.com/coreos/mantle/kola/tests/etcd"
 
 //register new tests here
 // "$name" and "$discovery" are substituted in the cloud config during cluster creation
-var Tests = []Test{
-	// test etcd discovery with 0.4.7
-	Test{
-		Run:         etcdDiscovery,
-		Discovery:   true,
-		ClusterSize: 3,
-		Name:        "etcd1 discovery",
-		EtcdVersion: 1,
-		CloudConfig: `#cloud-config
+func init() {
+	Tests = append(Tests, []Test{
+		// test etcd discovery with 0.4.7
+		Test{
+			Run:         etcd.DiscoveryV1,
+			ClusterSize: 3,
+			Name:        "etcd1 discovery",
+			Group:       "etcd",
+			CloudConfig: `#cloud-config
 coreos:
   etcd:
     name: $name
     discovery: $discovery
     addr: $public_ipv4:4001
     peer-addr: $private_ipv4:7001`,
-	},
+		},
 
-	// test etcd discovery with 2.0 with new cloud config
-	Test{
-		Run:         etcdDiscovery,
-		Discovery:   true,
-		ClusterSize: 3,
-		Name:        "etcd2 discovery",
-		EtcdVersion: 2,
-		CloudConfig: `#cloud-config
+		// test etcd discovery with 2.0 with new cloud config
+		Test{
+			Run:         etcd.DiscoveryV2,
+			ClusterSize: 3,
+			Name:        "etcd2 discovery",
```

Comment:
Don't use spaces in test names, that'll make it easier to specify them as a command line option.